web-app/src/components/DipMap.js

```diff
@@ -745,6 +745,110 @@ export default class DipMap extends React.Component {
 					this.debugCount("updateMap/hidProvince");
 				}
 			}
+
+
+			//Here we check each non-SC and non-Sea territory. If all surrounding SCs are of the same power and none is "Neutral", colour them that power.
+			//Get all nodes, disqualify Sea and SC, and per node collect the edges in an array.
+			for (let prov in this.state.variant.Properties.Graph.Nodes) {
+				const node = this.state.variant.Properties.Graph.Nodes[prov];
+				if (!node.SC && node.Subs[""].Flags.Land) {
+					const borderprovs = [];
+					for (let edge in node.Subs[""].Edges) {
+						const edgenode = this.state.variant.Properties.Graph.Nodes[edge];
+						if (typeof edgenode !== "undefined") {
+							if (edgenode.SC) {
+								borderprovs.push(SCs[edgenode.Name]);
+							}
+						}
+					}
+
+					//check if all members of array are equal/not neutral.
+					let shoulddraw = true;
+					for (let i = 0; i < borderprovs.length; i++) {
+						if (
+							borderprovs[i] !== borderprovs[0] ||
+							borderprovs[i] == "Neutral"
+						) {
+							shoulddraw = false;
+							break;
+						}
+					}
```

Comment:
We can refactor this nicely using `every`. also camel case!
```
const firstProv = borderProvs[0];
let shouldDraw = borderProvs.every((prov) => prov !== firstProv || prov === "Neutral");
```

Reply:
This would be nice, but it doesn't work. It looks to do the same thing, but:

![image](https://user-images.githubusercontent.com/7293081/132400965-0afb0bf4-33e4-479c-816b-3a0ade7f33b7.png)
![image](https://user-images.githubusercontent.com/7293081/132400994-3296ebc8-273f-4034-bc35-45c13aa8455e.png)

while 

![image](https://user-images.githubusercontent.com/7293081/132401030-aa50ec07-af22-4dd9-9f8d-bc268a1719f4.png)
![image](https://user-images.githubusercontent.com/7293081/132401086-e0de4542-d984-44ca-9144-8527a2186ec7.png)

(See York, LVN, etc)

Reply:
...but I also don't understand why it doesn't work...

Reply:
Ah I apologise. The correct code is: 
```
let shouldDraw = !borderProvs.every((prov) => prov !== firstProv || prov === "Neutral");
```
Notice exclamation mark